pkg/hubble/exporter/dynamic_exporter.go

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package exporter
+
+import (
+	"context"
+	"errors"
+
+	"github.com/sirupsen/logrus"
+
+	v1 "github.com/cilium/cilium/pkg/hubble/api/v1"
+	"github.com/cilium/cilium/pkg/hubble/exporter/exporteroption"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+type dynamicExporter struct {
+	FlowLogExporter
+	logger           logrus.FieldLogger
+	watcher          *configWatcher
+	managedExporters map[string]*managedExporter
+	maxFileSizeMB    int
+	maxBackups       int
+	// mutex protects from concurrent modification of managedExporters by config
+	// reloader when hubble events are processed
+	mutex lock.RWMutex
+}
+
+// OnDecodedEvent distributes events across all managed exporters.
+func (d *dynamicExporter) OnDecodedEvent(ctx context.Context, event *v1.Event) (bool, error) {
+	select {
+	case <-ctx.Done():
+		return false, d.Stop()
+	default:
+	}
+
+	d.mutex.RLock()
+	defer d.mutex.RUnlock()
+
+	var errs error
+	for _, me := range d.managedExporters {
+		if me.config.End == nil || me.config.End.After(time.Now()) {
+			_, err := me.exporter.OnDecodedEvent(ctx, event)
+			errs = errors.Join(errs, err)
+		}
+	}
+	return false, errs
+}
+
+// Stop stops configuration watcher  and all managed flow log exporters.
+func (d *dynamicExporter) Stop() error {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	d.watcher.Stop()
+
+	var errs error
+	for _, me := range d.managedExporters {
+		errs = errors.Join(errs, me.exporter.Stop())
+	}
+
+	return errs
+}
+
+// NewDynamicExporter creates instance of dynamic hubble flow exporter.
+func NewDynamicExporter(logger logrus.FieldLogger, configFilePath string, maxFileSizeMB, maxBackups int) *dynamicExporter {
```

Comment:
This function is public yet the dynamicExporter is private, that's a little odd. Do we want to make DynamicExporter public?

Reply:
I made it public